src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_help.py

```diff
@@ -562,11 +562,11 @@
                 --account-name $my_diagnostic_storage_account --expiry 2037-12-31T23:59:00Z \\
                 --permissions wlacu --resource-types co --services bt -o tsv)
 
-            protected_settings="{'storageAccountName': '{my_diagnostic_storage_account}', \\
-                'storageAccountSasToken': '{storage_sastoken}'}"
+            protected_settings="{'storageAccountName': '$my_diagnostic_storage_account', \\
+                'storageAccountSasToken': '$storage_sastoken'}"
 
-            az vm diagnostics set --settings "{default_config}" \\
-                --protected-settings "{protected_settings}" \\
+            az vm diagnostics set --settings "$default_config" \\
+                --protected-settings "$protected_settings" \\
```

Comment:
Have you tested this syntax to ensure it works (i.e. that the quote doesn't preserve the $ instead of evaluating it)?

Reply:
I tested this and it works. 

p.s.
Turns out the `$` is preserved and not treated as a shell variable expansion if it is in single quotes and not double quotes. Except if the single quotes are surrounded by double quotes as on this [line](https://github.com/Azure/azure-cli/pull/8117/files/d21f80286e20806cb57950686339be1696a09ed7#diff-fbc1ce052f43c94a205cae4904c4bd12R565)